hatchet/graphframe.py

```diff
@@ -313,40 +313,62 @@ def to_hdf(self, filename, key="hatchet_graphframe", **kwargs):
         HDF5Writer(filename).write(self, key=key, **kwargs)
 
     def copy(self):
-        """Return a shallow copy of the graphframe.
+        """Return a partially shallow copy of the graphframe.
+
+        This copies the DataFrame object, but the data is comprised of references. The Graph is shared between self and the new GraphFrame.
+
+        Arguments:
+            self (GraphFrame): Object to make a copy of.
 
-        This copies the DataFrame, but the Graph is shared between self and
-        the new GraphFrame.
+        Returns:
+            other (GraphFrame): Copy of self
+                graph (graph): Reference to self's graph
+                dataframe (DataFrame): Pandas "non-deep" copy of dataframe
+                exc_metrics (list): Copy of self's exc_metrics
+                inc_metrics (list): Copy of self's inc_metrics
+                default_metric (str): N/A
+                metadata (dict): Copy of self's metadata
         """
         return GraphFrame(
             self.graph,
-            self.dataframe.copy(),
-            list(self.exc_metrics),
-            list(self.inc_metrics),
+            self.dataframe.copy(deep=False),  # copy to prevent dataframe mutation operations from modifying the original.
+            copy.copy(self.exc_metrics),
+            copy.copy(self.inc_metrics),
             self.default_metric,
-            self.metadata,
+            copy.copy(self.metadata),
```

Comment:
Maybe the decision here should be between `copy.deepcopy` and no copy at all since the behavior may be unclear if any of these objects contain other objects.

Reply:
Looking into this further, it appears that this only matters in very specific cases. Say if one list contained a `hatchet.node.Node()` object and you changed the `_hatchet_nid` value. The value would be changed for the node in both `self` and `other`'s lists whereas this is not the case of `deepcopy` since the node would be a separate object. 

This is not a realistic concern for any of these attributes however, since none will contain any complex objects, so `copy.copy()` should suffice.